### PR TITLE
Feat/transaction notifications

### LINKED ISSUE #6516: In-app notification UI
As a user I would like to get notified when there is new transaction in order to increase my feeling of security that transaction has been proceed. 
As a user I would like to see what has happened since I was in the app in order to keep up to date. 

**Acceptance criteria:**
- When user is in the app and there is new pending transaction (incoming / outgoing) for imported XPUB asset, we display notification
- Notification duration is 0.2s to appear, then it stays for 5s, then disappears in 1s
- User can rid of notification by swiping up
- When user clicks on the notification, he is redirected to related transaction
- When transaction (incoming / outgoing) is confirmed , we display notification
- If transaction failed, we display notification

This task consists only implementing designs for all possible notifications types and just displaying them as standart "Box" on Demo screen. Whole logic of showing etc. will be another task.

This task should create new package `@suite-native/notifications`

Figma: https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/s2-Mobile?node-id=12256%3A96223&t=cIRdnBGEAEQ6DT1X-1
Components: https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/s2-Mobile?node-id=9143%3A54133&t=cIRdnBGEAEQ6DT1X-1
animations: https://www.figma.com/proto/Z6AGVUmKQzLNtDozFamW7f/branch/tCSmT0EOdXmhAst051Nxal/s2-Mobile?node-id=9926%3A67312&scaling=min-zoom&page-id=7888%3A55194&starting-point-node-id=9926%3A67312&show-proto-sidebar=1

```[tasklist]
### Tasks

### PR DESCRIPTION
**Pull request by PeKne**
## Related Issue

closes [#6516](https://github.com/trezor/trezor-suite/issues/6516)

## Screenshots:

https://user-images.githubusercontent.com/26143964/220591859-e6b08f2a-e0a4-4b21-9775-3ab471d2c4f9.mov

### CODE REVIEW

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}: TransactionIconSpinnerProps) => {
+    const { utils } = useNativeStyles();
+
+    const rotation = useTiming({ from: 0, to: 100, loop: true }, { duration: 2500 });
+    const transformRotate = useComputedValue(
+        () => [{ rotate: ((2 * Math.PI) / 100) * rotation.current }],
+        [rotation],
+    );
+    return (
+        <Box style={{ position: 'absolute' }}>
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: No inline styles

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+type TransactionIconSpinnerProps = {
+    radius: number;
+    color: CSSColor;
+    strokeWidth?: number;
+};
+
+export const TransactionIconSpinner = ({
+    radius,
+    color,
+    strokeWidth = 8,
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: Currently, you are not changing this value anywhere so I wouldn't pass it as optional prop. Just use it directly

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const { utils } = useNativeStyles();
+
+    const rotation = useTiming({ from: 0, to: 100, loop: true }, { duration: 2500 });
+    const transformRotate = useComputedValue(
+        () => [{ rotate: ((2 * Math.PI) / 100) * rotation.current }],
+        [rotation],
+    );
+    return (
+        <Box style={{ position: 'absolute' }}>
+            <Canvas style={{ height: radius * 2, width: radius * 2 }}>
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: No inline style

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        padding: 14.5,
+    }),
+);
+
+const cryptoIconStyle = prepareNativeStyle<TransactionIconStyleProps>((utils, { isAnimated }) => ({
+    position: 'absolute',
+    right: -2,
+    bottom: -2,
+    padding: 2,
+    backgroundColor: utils.colors[isAnimated ? 'gray1000' : 'gray0'],
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: Can you add comment to explain why this is needed? Not really sure why you need negative values, is this really necessary?

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        <Box>
+            <Box style={applyStyle(transactionIconStyle, { isAnimated })}>
+                <Icon
+                    name={transactionIconMap[transactionType]}
+                    color={isAnimated ? animatedIconColor : 'gray600'}
+                    size="mediumLarge"
+                />
+            </Box>
+            {isAnimated && (
+                <TransactionIconSpinner radius={ICON_SIZE / 2} color={animatedIconColor} />
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: I would like the component API more if we had something like IconSpinner or AnimatedIcon, which would wrap normal icon and add this TransactionIconSpinner. Both components are made with skia, so should be possible?

Comment (juriczech) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: It's weird that you need to send different color based on animation to normal icon and then as well render this TransactionIconSpinner

File: package.json [hunk 6]
```diff
@@ -37,7 +37,7 @@
         "lint-staged": "npx lint-staged",
         "lint:shellcheck": "./scripts/shellcheck.sh",
         "_______ Global Scripts _______": "Shared scripts for running in all workspaces",
-        "g:eslint": "cd $INIT_CWD && eslint --report-unused-disable-directives --cache --ignore-path ../../.gitignore",
+        "g:eslint": "cd $INIT_CWD && eslint --report-unused-disable-directives --ignore-path ../../.gitignore",
```
Comment (juriczech) [inline package.json]: Are you sure we want to push this? Could have a negative effect on CI if this script gets called there. I guess you verified it, just making sure 👌

File: suite-native/notifications/package.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        "@suite-native/navigation": "workspace:*",
+        "@suite-native/transactions": "workspace:*",
+        "@trezor/icons": "workspace:*",
+        "@trezor/styles": "workspace:*",
+        "@trezor/theme": "workspace:*",
+        "react": "18.2.0",
+        "react-native": "0.71.3",
+        "react-native-gesture-handler": "^2.9.0",
+        "react-native-reanimated": "2.14.4",
+        "react-redux": "7.2.2"
```
Comment (juriczech) [inline suite-native/notifications/package.json]: You should use version 8.0.5 or there will be version mismatch throughout the repo

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    selectTransactionFirstTargetAddress,
+} from '@suite-common/wallet-core';
+import {
+    RootStackRoutes,
+    StackNavigationProps,
+    RootStackParamList,
+} from '@suite-native/navigation';
+import { Icon } from '@trezor/icons';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { useNativeStyles, prepareNativeStyle } from '@trezor/styles/src';
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: just @trezor/styles

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+}));
+
+const transactionTypeToContentMap = {
+    recv: { title: 'Incoming transaction', preposition: 'from' },
+    sent: { title: 'Sending transaction', preposition: 'to' },
+    self: { title: 'Sending transaction', preposition: 'to' },
+    joint: { title: 'Sending transaction', preposition: 'to' },
+    failed: { title: 'Sending transaction', preposition: 'to' },
+    unknown: { title: 'Sending transaction', preposition: 'to' },
+} as const satisfies Record<TransactionType, TransactionTypeProperties>;
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: Are these titles correct? Failed transaction shouldn't display "Sending transaction"

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+export const TransactionNotificationDescription = ({
+    amount,
+    targetAddress,
+    transactionType,
+    networkSymbol,
+}: TransactionNotificationDescriptionProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { preposition } = transactionTypeToContentMap[transactionType];
+    return (
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: nit: \n

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        <Box flexDirection="row">
+            <CryptoAmountFormatter
+                value={amount}
+                network={networkSymbol}
+                isBalance={false}
+                color="gray500"
+                variant="label"
+            />
+            <Text color="gray500" variant="label">
+                {` ${preposition} `}
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: Why not just `{preposition}`?

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+        selectTransactionByTxidAndAccountKey(state, txid, accountKey),
+    );
+
+    const transactionTargetAddress = useSelector((state: TransactionsRootState) =>
+        selectTransactionFirstTargetAddress(state, txid, accountKey),
+    );
+
+    if (!transaction) return null;
+
+    const { title } =
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: Shouldn't return null if type is not in content map?

File: suite-native/notifications/src/components/AnnouncementNotification.tsx [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+            if (event.translationY < DISMISS_THRESHOLD) {
+                translateY.value = withTiming(-700, undefined, isFinished => {
+                    if (isFinished) runOnJS(setIsHidden)(true);
+                });
+            } else {
+                translateY.value = withTiming(0);
+            }
+        },
+    });
+
```
Comment (juriczech) [inline suite-native/notifications/src/components/AnnouncementNotification.tsx]: You can move this outside component so it doesn't get recreated on each rerender? Move to hook

File: packages/icons/src/components/Icon.tsx [hunk 14]
```diff
@@ -31,7 +31,8 @@ export const Icon = ({ name, customSize, size = 'large', color = 'gray1000' }: I
     } = useNativeStyles();
     const sizeNumber = customSize || iconSizes[size];
     const paint = useMemo(() => Skia.Paint(), []);
-    paint.setColorFilter(Skia.ColorFilter.MakeBlend(Skia.Color(colors[color]), BlendMode.SrcIn));
+    const skiaColor = color in colors ? colors[color as Color] : color;
```
Comment (juriczech) [inline packages/icons/src/components/Icon.tsx]: do you need `color as Color` since you've already narrowed this type with the condition?

File: suite-native/module-dev-utils/src/fixtures.ts [hunk 15]
```diff
@@ -0,0 +1,12 @@
+export const transactionNotifications = [
```
Comment (juriczech) [inline suite-native/module-dev-utils/src/fixtures.ts:1]: nit: if you're using it only in demo screen, you can move it there without this new file to have the least amount of imports there so it's easier to delete once we switch to some better solution for component demo, but it doesn't really matter that much

**Review by juriczech — CHANGES_REQUESTED** [review]

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (Nodonisko) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: I would prefer to fallback to uknown, null is problematic because it's very hard to debug if user report something strange.

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 3 above)
Comment (Nodonisko) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: We agreed that it's OK for Skia only components, if it's simple style.

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 5 above)
Comment (Nodonisko) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: Can you give example of API? I am not sure if I am understand.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (juriczech) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: Yes, or use the Error message from atoms, that might be best

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 5 above)
Comment (juriczech) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: Just a simple composition like this. Maybe it's not possible, that's why I'm asking. But since both components are skia, it should be possible, right?

```
<TransactionIconSpinner>
  <Icon {...props} />
</TransactionIconSpinner>
```

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 3 above)
Comment (juriczech) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: Okay, I didn't know about that. I thought we don't want to use them ever as you commented on some PR before

File: suite-native/module-dev-utils/src/fixtures.ts (hunk 15 above)
Comment (PeKne) [reply suite-native/module-dev-utils/src/fixtures.ts:1]: I did it because I think that the demo screen file is already too bloated.

File: packages/icons/src/components/Icon.tsx (hunk 14 above)
Comment (PeKne) [reply packages/icons/src/components/Icon.tsx]: Not sure why, but yes I do.
Screenshot 2023-02-22 at 13 30 48

File: suite-native/notifications/src/components/AnnouncementNotification.tsx (hunk 13 above)
Comment (PeKne) [reply suite-native/notifications/src/components/AnnouncementNotification.tsx]: It is already a hook, so it does not recreate. Thus I don't see any reason to isolate it in another file since this is the only place where it will be used.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: There is no way that type will be missing. The object is correctly typed. If there would be any of the transaction types missing, the linter would shout. example:

https://user-images.githubusercontent.com/26143964/220620046-16817efa-979c-43f7-8e45-79a81654e5cb.mov

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 11 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: because there would be no whitespace around. The result would look like e.g `0.003 BTCfromdsjf...fskdfd`

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 5 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: What is the benefit of this solution? IMHO it would make the component only more complicated. Actually, I'm not even sure if it's possible to draw one skia `Canvas` inside of another one.

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 4 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: To follow the design, the cryptoIcon should overflow the spinner by two "pixels" ([see here](https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/s2-Mobile?node-id=12242%3A92736&t=Es3ANdXAW24XIKUP-4)). With the combination of `position: absolute` the negative bottom and right achieves that result.

Is there any general rule that negative values should be never used?

File: package.json (hunk 6 above)
Comment (PeKne) [reply package.json]: This is a mistake, already fixed it. Will be part of a fixup commit. But good catch. That would definitely affect CI performance.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 10 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: fixed in 11399f17d

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 8 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: fixed 11399f17d

File: suite-native/notifications/package.json (hunk 7 above)
Comment (PeKne) [reply suite-native/notifications/package.json]: fixed 11399f17d

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 1 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: fixed 11399f17d

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 2 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: fixed 11399f17d

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 5 above)
Comment (Nodonisko) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: It is not possible to render canvas inside another canvas. Also if I understand it correctly I don't see any befits of this solution that would justify that added  complexity and any future customization will be much harder.

File: package.json (hunk 6 above)
Comment (Nodonisko) [reply package.json]: Not really this could affect only local performance, there two caches one in Nx which will work independently and second in eslint which is one you disabled. Eslint cache is not used on CI because files for this cache is gitignored. But yea, definitely better to keep this on :D

File: packages/icons/src/components/Icon.tsx [hunk 16]
```diff
@@ -3,15 +3,15 @@ import React, { useMemo } from 'react';
...
+import { Color, CSSColor } from '@trezor/theme';
 
 import { IconName, icons } from '../icons';
 
 type IconProps = {
     name: IconName;
     size?: IconSize;
     customSize?: number;
-    color?: Color;
+    color?: Color | CSSColor;
```
Comment (Nodonisko) [inline packages/icons/src/components/Icon.tsx]: Why do we need this? We should probably never use color that is not our theme pallete.

**Review by Nodonisko — CHANGES_REQUESTED** [review]
Will do full review later.

File: packages/icons/src/components/Icon.tsx (hunk 16 above)
Comment (PeKne) [reply packages/icons/src/components/Icon.tsx]: Ideally, we should not use any colour out of our colour theme. But this new notification design is using "inverted" theme. That means that if the app theme is set to "standard" the displayed notification will use colours from the dark theme and vice versa. 

For this reason, I need to extract the hex value of the `dark theme yellow` to the white colour theme.

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 5 above)
Comment (juriczech) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: Well the benefit is that when you want to use it in render, you just render 1 thing I guess. And it would be easier to reuse if we ever used it with other icon types (highly unlikely though). 

But doesn't matter, since you cannot render canvas inside canvas.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 11 above)
Comment (juriczech) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: I was going to say to use marginHorizontal, but I guess it's easier like this

File: suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx (hunk 4 above)
Comment (juriczech) [reply suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx]: Not really. Generally I just pay more attention to it as 95% of the time it's not needed and just signals bad styling. Fine here I guess.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (juriczech) [reply suite-native/notifications/src/components/TransactionNotification.tsx:109]: Yes, so you can remove the fallback then 😉

Since there is need to have a fallback, either use unknown or show error. I know that currently all types are supported, but it won't hurt to be on the safe side. I guess the fallback is never going to be used anyway if I use your argument, we should remove it then 😅

If you just fallback to unknown, it would make sense, but it doesn't make sense to fallback to `recv`, that's just wrong.

File: packages/icons/src/components/Icon.tsx (hunk 14 above)
Comment (juriczech) [reply packages/icons/src/components/Icon.tsx]: Maybe if you showed the error, it would be more helpful 😅

I checked it out in the branch and while I was there I pushed the change. If you think it's unnecessary, revert it but it gets rid of the type casting. Not sure whether it's the best way to do it. 
https://github.com/trezor/trezor-suite/pull/7685/commits/03bbc5e776f541f87b83e3f44bebd4a1579e8d51

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (juriczech) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: Yeah. Since it's typed correctly, it doesn't need the fallback case. If you remove it, typescript doesn't scream

```suggestion
    const { title } = transactionTypeToContentMap[transaction.type];
```

File: suite-native/module-dev-utils/src/fixtures.ts (hunk 15 above)
Comment (juriczech) [reply suite-native/module-dev-utils/src/fixtures.ts:1]: Yeah, it's not too pretty 😀

**Comment by PeKne** [comment]
I have added `InvertedThemeProvider` to handle the inverted colour theme issue. @Nodonisko @juriczech please review again 🙏.

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 12 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: fixed [0dfad91](https://github.com/trezor/trezor-suite/commit/0dfad91c31f18abfb3a328936724ab34c064a83d)

File: packages/icons/src/components/Icon.tsx (hunk 14 above)
Comment (PeKne) [reply packages/icons/src/components/Icon.tsx]: The whole colour definition was reworked. Not relevant anymore.

File: packages/icons/src/components/Icon.tsx (hunk 16 above)
Comment (PeKne) [reply packages/icons/src/components/Icon.tsx]: fixed in 0dfad91

File: suite-native/theme/src/useGetSystemColorVariant.ts [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+
+import { ThemeColorVariant } from '@trezor/theme';
+
+export const useGetSystemColorVariant = (): ThemeColorVariant => {
+    const colorScheme = useColorScheme();
+    if (colorScheme === 'dark') {
+        return 'dark';
+    }
+    return 'standard';
+};
```
Comment (juriczech) [inline suite-native/theme/src/useGetSystemColorVariant.ts:11]: Why is this necessary? Basically it enforces only `dark` / `standard` theme in inverted theme provider?

Comment (juriczech) [reply suite-native/theme/src/useGetSystemColorVariant.ts:11]: Seems quite redundent at this moment

**Review by juriczech — APPROVED** [review]

File: suite-native/notifications/src/components/AnnouncementNotification.tsx [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+    const translateY = useSharedValue(0);
+
+    const onSwipeGesture = useAnimatedGestureHandler<PanGestureHandlerGestureEvent>({
+        onActive: event => {
+            if (event.translationY <= 0) translateY.value = event.translationY;
+        },
+        onEnd: event => {
+            if (event.translationY < DISMISS_THRESHOLD) {
+                translateY.value = withTiming(-700, undefined, isFinished => {
```
Comment (Nodonisko) [inline suite-native/notifications/src/components/AnnouncementNotification.tsx]: What is `-700`? Some well named constant that would explain that could be nice.

File: suite-native/notifications/src/components/AnnouncementNotification.tsx [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+                    if (isFinished) runOnJS(setIsHidden)(true);
+                });
+            } else {
+                translateY.value = withTiming(0);
+            }
+        },
+    });
+
+    useEffect(() => {
```
Comment (Nodonisko) [inline suite-native/notifications/src/components/AnnouncementNotification.tsx]: `isHiddenAutomatically` missing in useEffect deps

File: suite-native/notifications/src/components/AnnouncementNotification.tsx [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingVertical: utils.spacings.small / 2,
+    borderRadius: utils.borders.radii.round,
+    backgroundColor: utils.colors.gray1000,
+    paddingHorizontal: utils.spacings.small / 2,
+    ...utils.boxShadows.small,
+}));
+
+export const AnnouncementNotification = ({
```
Comment (Nodonisko) [inline suite-native/notifications/src/components/AnnouncementNotification.tsx]: I am not sure about name, why `Annoucement`? I would prefer something like just `Notification` or `ToastNotification`

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+    title?: string;
+    preposition?: string;
+};
+
+const addressContainerStyle = prepareNativeStyle(_ => ({
+    maxWidth: '35%',
+}));
+
+const transactionTypeToContentMap = {
+    recv: { title: 'Incoming transaction', preposition: 'from' },
```
Comment (Nodonisko) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: nit: `preposition` sounds really strange :D

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+const STROKE_WIDTH = 8;
+
+const ContainerStyle = prepareNativeStyle(_ => ({
+    position: 'absolute',
+}));
+
+export const TransactionIconSpinner = ({ radius, color }: TransactionIconSpinnerProps) => {
+    const { applyStyle, utils } = useNativeStyles();
+
+    const rotation = useTiming({ from: 0, to: 100, loop: true }, { duration: 2500 });
```
Comment (Nodonisko) [inline suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: Please use Reanimated instead https://shopify.github.io/react-native-skia/docs/animations/reanimated/

We will have to migrate it soon anyway because Skia will support reanimated directly.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 22 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: fixed in [40bdfc5](https://github.com/trezor/trezor-suite/pull/7685/commits/40bdfc547bec9d4972f293bc2fa699868a835f00)

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 21 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: renamed to `prefix` in [40bdfc5](https://github.com/trezor/trezor-suite/pull/7685/commits/40bdfc547bec9d4972f293bc2fa699868a835f00)

File: suite-native/notifications/src/components/AnnouncementNotification.tsx (hunk 19 above)
Comment (PeKne) [reply suite-native/notifications/src/components/AnnouncementNotification.tsx]: fixed in [40bdfc5](https://github.com/trezor/trezor-suite/pull/7685/commits/40bdfc547bec9d4972f293bc2fa699868a835f00)

File: suite-native/notifications/src/components/AnnouncementNotification.tsx (hunk 18 above)
Comment (PeKne) [reply suite-native/notifications/src/components/AnnouncementNotification.tsx]: fixed in [40bdfc5](https://github.com/trezor/trezor-suite/pull/7685/commits/40bdfc547bec9d4972f293bc2fa699868a835f00)

File: suite-native/notifications/src/components/AnnouncementNotification.tsx (hunk 20 above)
Comment (PeKne) [reply suite-native/notifications/src/components/AnnouncementNotification.tsx]: The naming was based on the design Figma file. But I agree it's not a very accurate name. Renamed to `Notification` in 2f14645f6.

File: suite-native/theme/src/useUserColorScheme.ts [hunk 23]
```diff
@@ -0,0 +1,17 @@
+import { useAtom } from 'jotai';
+
+import { ThemeColorVariant } from '@trezor/theme';
+import { atomWithMMKV } from '@suite-native/storage';
+
+export type AppColorScheme = ThemeColorVariant | 'system';
+
+const userColorSchemeAtom = atomWithMMKV<AppColorScheme>('colorScheme', 'standard');
```
Comment (Nodonisko) [inline suite-native/theme/src/useUserColorScheme.ts]: I think you changed logic here little bit, it should set dark theme as default in case device is using dark mode. I think if you will set this `system` it should work as before.

File: suite-native/storage/src/storage.ts [hunk 24]
```diff
@@ -24,14 +27,17 @@ const retrieveStorageEncryptionKey = async () => {
     return secureKey;
 };
 
-export const initMmkvStorage = async (): Promise<Storage> => {
```
Comment (Nodonisko) [inline suite-native/storage/src/storage.ts:27]: nit: not sure if there is benefit of spliting this to two functions since it again used only here

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/notifications/src/components/TransactionNotification.tsx [hunk 25]
```diff
@@ -0,0 +1,143 @@
...
+const transactionTypeToContentMap = {
+    recv: { title: 'Incoming transaction', prefix: 'from' },
+    sent: { title: 'Sending transaction', prefix: 'to' },
+    self: { title: 'Sending transaction', prefix: 'to' },
+    joint: { title: 'Sending transaction', prefix: 'to' },
+    failed: { title: 'Sending transaction', prefix: 'to' },
+    unknown: { title: 'Sending transaction', prefix: 'to' },
+} as const satisfies Record<TransactionType, TransactionTypeProperties>;
+
+export const TransactionNotificationDescription = ({
```
Comment (juriczech) [inline suite-native/notifications/src/components/TransactionNotification.tsx]: nit: i would move this to separate file

File: suite-native/storage/src/storage.ts [hunk 26]
```diff
@@ -47,3 +53,26 @@ export const initMmkvStorage = async (): Promise<Storage> => {
...
+    id: MMKV_STORAGE_ID,
+});
+
+export const atomWithMMKV = <T>(atomKey: string, initialValue: T) => {
+    const JotaiMMKVStorage = createJSONStorage<T>(() => ({
+        getItem: (key: string) => {
+            const value = unecryptedJotaiStorage.getString(key);
+            return value ? JSON.parse(value) : null;
+        },
+        setItem: (key: string, value: unknown): void => {
```
Comment (juriczech) [inline suite-native/storage/src/storage.ts]: isn't value `T` generic?

File: suite-native/theme/src/InvertedThemeProvider.tsx [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+import { useUserColorScheme } from './useUserColorScheme';
+
+type InvertedThemeProviderProps = {
+    children: ReactNode;
+};
+
+export const InvertedThemeProvider = ({ children }: InvertedThemeProviderProps) => {
+    const { userColorScheme } = useUserColorScheme();
+    const systemColorScheme = useGetSystemColorScheme();
+    const activeColorScheme = userColorScheme === 'system' ? systemColorScheme : userColorScheme;
```
Comment (juriczech) [inline suite-native/theme/src/InvertedThemeProvider.tsx]: Why not have all of this within one hook?

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+        [rotation.value],
+    );
+
+    useEffect(() => {
+        rotation.value = withRepeat(
+            withTiming(360, {
+                duration: 2500,
+                easing: Easing.linear,
+            }),
+            -1,
```
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: nit: Please use constants as in other files. At least for the `-1` value so it's readable from const name what it is without having to think about it too much

File: suite-native/theme/src/useUserColorScheme.ts (hunk 23 above)
Comment (PeKne) [reply suite-native/theme/src/useUserColorScheme.ts]: fixed ec7bdb40a

File: suite-native/storage/src/storage.ts (hunk 24 above)
Comment (PeKne) [reply suite-native/storage/src/storage.ts:27]: fixed ec7bdb40a

File: suite-native/notifications/src/components/TransactionNotification.tsx (hunk 25 above)
Comment (PeKne) [reply suite-native/notifications/src/components/TransactionNotification.tsx]: moved in ec7bdb40a

File: suite-native/storage/src/storage.ts (hunk 26 above)
Comment (PeKne) [reply suite-native/storage/src/storage.ts]: good catch. Changed in ec7bdb40a

File: suite-native/theme/src/InvertedThemeProvider.tsx (hunk 27 above)
Comment (PeKne) [reply suite-native/theme/src/InvertedThemeProvider.tsx]: Good idea ec7bdb40a

File: suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx (hunk 28 above)
Comment (PeKne) [reply suite-native/transactions/src/components/TransactionsList/TransactionIconSpinner.tsx]: rename in ec7bdb40a

File: suite-native/storage/src/atomWithMMKV.ts [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+
+function removeItem(key: string): void {
+    unecryptedJotaiStorage.delete(key);
+}
+
+function clearAll(): void {
+    unecryptedJotaiStorage.clearAll();
+}
+
+export const atomWithMMKV = <T>(key: string, initialValue: T) =>
```
Comment (Nodonisko) [inline suite-native/storage/src/atomWithMMKV.ts]: Please rename this to `atomWithUnecryptedStorage` to be clear about fact that this is not encrypted storage so no one will misuse this.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

**Review by juriczech — APPROVED** [review]

File: suite-native/storage/src/atomWithMMKV.ts (hunk 29 above)
Comment (PeKne) [reply suite-native/storage/src/atomWithMMKV.ts]: renamed in af5661e4f

**Review by Nodonisko — APPROVED** [review]
